### ISSUE TITLE
Better integration with Angular migrations

### ISSUE THREAD

**Issue by GreedyA1** [nrwl/nx#13733]
### Current Behavior

when running angular material + cdk migrations on angular non-buildable libraries are skipped.

This issue is detailed well by [luchsamapparat](https://github.com/luchsamapparat) in [issue](https://github.com/nrwl/nx/issues/4262#issuecomment-754761356).

also created similar [issue ](https://github.com/angular/components/issues/26224)on angular side for better communication.

### Expected Behavior

* nx needs to add tsconfig to test target for non-buildable libraries in project with @nrwl/angular:library generator

or 

* if tsconfig is unecessary, there needs to be a better solution found with angular team.

### Steps to Reproduce

1. Create project in nx and angular 14 
2. run `nx migrate`
3. run `nx migrate --run-migrations`

find libraries skipping migrations because tsconfig wasn't found.

### Nx Report

```shell
nx : 15.0.0
   @nrwl/angular : 15.0.0
   @nrwl/cypress : 15.0.0
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.0.0
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.0.0
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.0.0
   @nrwl/js : 15.0.0
   @nrwl/linter : 15.0.0
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.0.0
   @nrwl/web : Not Found
   @nrwl/webpack : 15.0.0
   @nrwl/workspace : 15.0.0
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
         @ngrx/component-store: 14.3.2
         @ngrx/effects: 14.0.2
         @ngrx/entity: 14.0.2
         @ngrx/router-store: 14.0.2
         @ngrx/store: 14.0.2
         rxjs: 6.6.7
         @ngrx/schematics: 14.0.2
         @ngrx/store-devtools: 14.0.2
         @storybook/angular: 6.5.13
         ng-mocks: 14.3.1
```

**Comment by leosvelperez** [comment]
Thanks for reporting this!

We are aware of the issue and are in touch with the Angular team to find a solution.

Please note that in some cases, there's no real issue apart from the noisy logs. The runtime code of the non-buildable libs in an Nx workspace is actually processed as part of any app that consumes them. The Angular Material/CDK migrations create a TS program based on the tsconfig of any project. The files included in the TS program are then migrated. In the case of an Nx workspace with an app consuming non-buildable library, the source code of the non-buildable libraries is actually part of the TS program for the app and therefore, the migrations run over those files.

That said, some migrations (like v14 -> v15) also make changes to test files. Since the `test` targets in Nx workspaces normally use the `@nrwl/jest:jest` executor and the `tsconfig` option is not set in the project configuration, no test code is processed and migrated.

So, to summarize the current state:

- The runtime code of non-buildable libs is migrated correctly despite the warning
- The test code of apps and libs is not migrated (not all migrations need to migrate test code, but some do)

For now, a workaround would be to provide the deprecated `tsConfig` option for the `test` target (for projects using Jest, projects using Karma don't have an issue). A small script could even be run before and after the migrations to set and unset the option. Definitely not ideal, but it's the workaround for now.

We'll keep this open, so we can track any development on this front and provide updates.

**Comment by Coly010** [comment]
Until we get any further information on this front, and as there is a workaround provided in the message above, I'm going to close this issue. However, it can be reopened when more information comes through.
